packages/flutter_tools/lib/src/migrate/migrate_utils.dart

```diff
@@ -0,0 +1,326 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+import 'dart:typed_data';
+
+import '../base/common.dart';
+import '../base/file_system.dart';
+import '../base/logger.dart';
+import '../base/terminal.dart';
+import '../commands/migrate.dart';
+import '../globals.dart' as globals;
+
+// TODO(garyq): support windows.
+
+/// Utility class that contains static methods that wrap git and other shell commands.
+class MigrateUtils {
+  MigrateUtils();
+
+  /// Creates a temporary directory.
+  ///
+  /// This directory should be deleted after use.
+  static Future<Directory> createTempDirectory(String name) async {
```

Comment:
Can you just use `systemTempDirectory.createTempSync` as the rest of the tool does?

Reply:
Ahh sure, didn't know about this utility!